src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/iOS/WebUI.cs

```diff
@@ -118,9 +112,15 @@ public void Authenticate(Uri authorizationUri, Uri redirectUri, CallState callSt
         private void CallbackMethod(AuthorizationResult result)
         {
             SetAuthorizationResult(result);
-            inBackground = false;
-            DidEnterBackgroundNotification.Dispose();
-            WillEnterForegroundNotification.Dispose();
```

Comment:
Moving notification registration to new dispose method that is called when the webUI is closed.